### PR TITLE
Add source data removal scripts

### PR DESCRIPTION
**Pull request by whd**
Copy of https://github.com/mozilla-services/dataops/pull/20

The intermediate results and listings are in s3://net-mozaws-prod-us-west-2-pipeline-analysis/whd/bug_1550805/ (warning, ~10GB of data). main.sh is the theoretical script you would use to run each of these pieces in succession, and delete.sh is the script that you would run to perform the actual deletion.

Landfill buffering has a maximum time frame of 15 minutes, while decoded data has an hour.

`reprocess-structured-landfill/` is the hindsight config used to generate reprocessed structured landfill. Since it's landfill data the reprocessed version is being stored in a temporary prod bucket not accessible without ops credentials.

It would probably not be hard to use a similar config as above for the data identified as "edge cases" to have clean hour boundaries, but it would require reprocessing of about 500GB of data and we've decided it's ok to delete a bit of extra data.

NOTE: this doesn't create a backup, I will need to add scripts for that.

### CODE REVIEW

File: 2019-05-14-armagaddon/heka/generate_batches [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env python
+# generate s3api delete object batches
+# https://docs.aws.amazon.com/cli/latest/reference/s3api/delete-objects.html
+import sys
+import json
+skeleton = '''{"Objects": [{"Key": "test1.txt"}],"Quiet": false}'''
+j = json.loads(skeleton)
```
Comment (relud) [inline 2019-05-14-armagaddon/heka/generate_batches]: why is this pumped through the json decoder?
```suggestion
j = {"Objects": [{"Key": "test1.txt"}],"Quiet": False}
```

Comment (whd) [reply 2019-05-14-armagaddon/heka/generate_batches]: I was just using https://docs.aws.amazon.com/cli/latest/reference/s3api/delete-objects.html#examples precisely but it doesn't really matter either way.

File: 2019-05-14-armagaddon/heka/generate_batches [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+def chunks(l, n):
+    """Yield successive n-sized chunks from l."""
+    for i in xrange(0, len(l), n):
+        yield l[i:i + n]
+
+
+f = open(sys.argv[1], 'r').readlines()
+i = 0
+for chunk in chunks(f, 1000):
```
Comment (relud) [inline 2019-05-14-armagaddon/heka/generate_batches]: python has a function for adding an index to your for loop
```suggestion
for i, chunk in enumerate(chunks(f, 1000)):
```

File: 2019-05-14-armagaddon/heka/generate_batches [hunk 3]
```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env python
+# generate s3api delete object batches
+# https://docs.aws.amazon.com/cli/latest/reference/s3api/delete-objects.html
+import sys
+import json
+skeleton = '''{"Objects": [{"Key": "test1.txt"}],"Quiet": false}'''
```
Comment (relud) [inline 2019-05-14-armagaddon/heka/generate_batches]: not referenced anywhere
```suggestion
```

File: 2019-05-14-armagaddon/heka/generate_batches [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+f = open(sys.argv[1], 'r').readlines()
+i = 0
+for chunk in chunks(f, 1000):
+    objects = []
+    for line in chunk:
+        objects.append({"Key": (line.rstrip().split(' ')[-1])})
+    j["Objects"] = objects
+    with open("batch{}.json".format(i), "w") as f:
+        f.write(json.dumps(j))
+    i = i + 1
```
Comment (relud) [inline 2019-05-14-armagaddon/heka/generate_batches]: remove this line if enumerate is merged
```suggestion
```

File: 2019-05-14-armagaddon/heka/generate_batches [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+
+def chunks(l, n):
+    """Yield successive n-sized chunks from l."""
+    for i in xrange(0, len(l), n):
+        yield l[i:i + n]
+
+
+f = open(sys.argv[1], 'r').readlines()
+i = 0
```
Comment (relud) [inline 2019-05-14-armagaddon/heka/generate_batches]: remove this line if enumerate is merged
```suggestion
```

File: 2019-05-14-armagaddon/heka/generate_batches [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+
+f = open(sys.argv[1], 'r').readlines()
+for i, chunk in enumerate(chunks(f, 1000)):
+    objects = []
+    for line in chunk:
+        objects.append({"Key": (line.rstrip().split(' ')[-1])})
+    j["Objects"] = objects
+    with open("batch{}.json".format(i), "w") as f:
+        f.write(json.dumps(j))
```
Comment (relud) [inline 2019-05-14-armagaddon/heka/generate_batches]: this f is shadowing the one used to create the initial for loop, it should be fine, but I think it's worth renaming the variable for safety

**Review by relud — APPROVED** [review]

File: 2019-05-14-armagaddon/heka/reprocess-structured-landfill/hindsight/run/output/telemetry_s3.cfg.tmp [hunk 7]
```diff
@@ -0,0 +1,32 @@
+-- -*- lua -*-
+filename = "moz_telemetry_s3.lua"
+
+-- match data outside the affected period or data that isn't activity-stream
+message_matcher = "Timestamp < 1556967600000000000 || Timestamp > 1557572400000000000 || (Fields[uri] !~ '/submit/activity%-stream')"
```
Comment (trink) [inline 2019-05-14-armagaddon/heka/reprocess-structured-landfill/hindsight/run/output/telemetry_s3.cfg.tmp:5]: You can use the human readable timestamps (but the above appear to be correct) http://mozilla-services.github.io/lua_sandbox/util/message_matcher.html

File: 2019-05-14-armagaddon/heka/reprocess-structured-landfill/script.sh [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+# based on fields, and submission date (derived from Timestamp) isn't a field
+# for raw messages.
+for date in $(seq 20190504 20190511) ; do
+  sed "s/FIXME/$date/" landfill.json.tmp > landfill.json
+  sed "s/FIXME/$date/" hindsight/run/output/telemetry_s3.cfg.tmp > hindsight/run/output/telemetry_s3.cfg
+  cd hindsight
+  hindsight_cli hindsight.cfg 7
+  cd ..
+  rm -rf /media/ephemeral0/hindsight_output
+  rm -f hindsight/run/input/landfill*cfg
```
Comment (trink) [inline 2019-05-14-armagaddon/heka/reprocess-structured-landfill/script.sh:18]: The tmp directory could also be wiped but it shouldn't cause any problems if it isn't

**Review by trink — APPROVED** [review]
The HS bits look good

**Comment by whd** [comment]
I've added the complement to the files being deleted here to `s3://net-mozaws-prod-us-west-2-pipeline-analysis/whd/bug_1550805/keep/` for the consideration of @relud or @robotblake (size 800MB).

**Pull request merged**